### DISCUSSION TITLE
ladder logic and g code ,power arm ,3d printer

### DISCUSSION THREAD

**Discussion by roboo7** [vercel/next.js#52802] [Help] [answered]
### Summary

i watn your  attention i need your  help 
 i want to make mini power arm and modified 3d printer 
so i want to know how to program all things and also i know about ladder logic 
and g code include aurdino ,plc  i need all detail 

### Additional information

```js
like what is plc how to program and also program software 
diffrance between plc and aurdino how can i program power arm
```

**Comment by icyJoseph** [comment, marked as answer]
This is off-topic, but as luck would have it I've used PLCs at one point of my life. For a good chunk of it.

Ladder logic is usually programmed through a GUI using blocks, which can be interfaced with I/O and analog ports on PLCs. Most PLC manufacturers have their own software. I've used ABB, Siemens, Yokogawa and a couple of others that escape me. Often these IDE's have a place to write C like code to do more complex stuff that ladder's can't, or to implement custom blocks.

You can for example, read inputs from buttons and sensors, to control a motor, all through the PLC.

I've never used Arduino with ladder logic though. I prefer to use the actual Arduino language for that. 

A PLC is production solution that can be integrated into an industrial network protocol such as Profibus, HART, etc, it is often the heart of the network, or part of it, and it can receive/send data from/to the network through the protocol. Based on the data it collects, you can calculate what to do send back to actuators and what not. You can also do monitoring through a PLC.

Arduino is a micro-controller, which, through a lot of effort, you could turn into the heart of a PLC. 

Please try to keep this to the Next.js framework though.
